packages/headless/src/features/pagination/pagination-slice.test.ts

```diff
@@ -178,4 +182,49 @@ describe('pagination slice', () => {
       expect(finalState.numberOfResults).toEqual(state.numberOfResults);
     });
   });
+
+  describe('restore pagination on  facet toggle actions', () => {
+    it('when on page 2, and a facet is toggled, #firstResult is set to 0', () => {
+      state.firstResult = 10;
+      state.numberOfResults = 10;
+      expect(determinePage(state)).toBe(2);
+
+      const nextState = paginationReducer(state, toggleSelectFacetValue);
+
+      expect(nextState.firstResult).toEqual(0);
+    });
+
+    it('when on page 3, and a date facet is toggled, #firstResult is set to 0', () => {
+      state.firstResult = 20;
+      state.numberOfResults = 10;
+      expect(determinePage(state)).toBe(3);
+
+      const nextState = paginationReducer(state, toggleSelectDateFacetValue);
+
+      expect(nextState.firstResult).toEqual(0);
+    });
+
+    it('when on page 2, and a category facet is toggled, #firstResult is set to 0', () => {
+      state.firstResult = 10;
+      state.numberOfResults = 10;
+      expect(determinePage(state)).toBe(2);
+
+      const nextState = paginationReducer(
+        state,
+        toggleSelectCategoryFacetValue
+      );
+
+      expect(nextState.firstResult).toEqual(0);
+    });
+
+    it('when on page 4, and a numeric facet is toggled, #firstResult is set to 0', () => {
```

Comment:
I don't think testing with different pages is super relevant, but testing different actions is.
This is then a very good opportunity for creating a method that will take care of all the setup & excepts, taking only an action as parameter.

Quick example:
```
function testResetPagination(action: Action) {
    state.firstResult = 10;
    state.numberOfResults = 10;
    const nextState = paginationReducer(
        state,
        action
     );

      expect(nextState.firstResult).toEqual(0);
}
```

Reply:
I see, I wanted to simulate my manual tests but I guess it doesn't matter on which page the user is.

Reply:
Yes the important is the reset to 0, the value it resets from doesn't really matter. If it was a decrement, that would be case where it mattered when it comes to testing.